torch/csrc/distributed/c10d/Utils.hpp

```diff
@@ -521,30 +522,30 @@ using SizeType = uint64_t;
         continue;                                                         \
       } else if (                                                         \
           errno_local == WSAETIMEDOUT || errno_local == WSAEWOULDBLOCK) { \
-        TORCH_CHECK(false, "Socket Timeout");                             \
+        C10_THROW_ERROR(DistNetworkError, "Socket Timeout");              \
```

Comment:
any reason to use `C10_THROW_ERROR` over `TORCH_CHECK`?

Reply:
I feel its simpler with `C10_THROW_ERROR`, `TORCH_CHECK(false,...)` is a bit weird due to the `false` parameter. Also, doesn't `TORCH_CHECK(` always throw `c10:Error`? We need to throw `DistNetworkError` here.